com.unity.ml-agents/Runtime/Actuators/IActuator.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using Unity.MLAgents.Policies;
+
+namespace Unity.MLAgents.Actuators
+{
+    /// <summary>
+    /// Abstraction that facilitates the execution of actions.
+    /// </summary>
+    public interface IActuator
+    {
+        float[] Actions
+        {
+            get;
+        }
+
+        void UpdateActions(float[] fullActionBuffer, int offset);
+
+        /// <summary>
+        /// Get the number of actions this IActuator will execute on.
+        /// </summary>
+        int GetNumberOfActions();
+
+        /// <summary>
+        /// Get the sizes of each discrete branch if they exist.  The length of this
+        /// array is equal to what is returned from <see cref="GetNumberOfActions"/>.
+        /// </summary>
+        /// <returns></returns>
+        int[] GetBranchSizes();
+
+        void ResetData();
+
+        /// <summary>
+        ///  This method is called in order to allow the user execution actions
+        /// with the array of actions passed in.
+        /// </summary>
+        void OnActionReceived();
+
+        /// <summary>
+        /// Collects masks for discrete actions, please refer to
+        /// <see cref="Agent.CollectDiscreteActionMasks"/>.
+        /// </summary>
+        /// <param name="actionMasker"></param>
+        void CollectDiscreteActionMasks(DiscreteActionMasker actionMasker);
```

Comment:
We need to call this (and aggregate the masks) somewhere, right?

Reply:
yes, the bookkeeping for this isn't done yet.